examples/src/main/java/com/squareup/moshi/recipes/ReadJsonListKt.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi.recipes
+
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import com.squareup.moshi.adapter
+import com.squareup.moshi.recipes.models.Card
+
+internal class ReadJsonListKt {
+
+  //language=JSON
+  private val jsonString = """
+    [{"rank": "4",
+       "suit": "CLUBS"
+     },
+     {"rank": "A",
+      "suit": "HEARTS"
+      },
+     {"rank": "J",
+      "suit": "SPADES"
+     }]
+  """.trimIndent()
+
+  @OptIn(ExperimentalStdlibApi::class)
+  fun readJsonList() {
+    val moshi = Moshi.Builder().build()
+    val jsonAdapter = moshi.newBuilder()
+      .add(Types.supertypeOf(Card::class.javaObjectType), moshi.adapter<Card>())
```

Comment:
There's a reified extension for this case as well

Reply:
Actually I'm not sure why you're doing the wildcard setup here at all. Can you elaborate?

Reply:
I followed [this](https://github.com/square/moshi/blob/d7b70637d218189c1618c7d6389ae9197d28ffd0/moshi-kotlin-tests/src/test/kotlin/com/squareup/moshi/kotlin/DualKotlinTest.kt#L462)